docs/amplitude/README.md

```diff
@@ -0,0 +1,117 @@
+
+The destination project is determined by which API key we use when making HTTP
+calls to Amplitude. The job looks for the API key as an environment variable,
+and it's generally the responsibility of the data engineer helping to deploy your
+job to make sure the job is invoked with the right API key for your project.
+
+## Schema for JSON Configuration Files
+
+The schema for JSON configuration files is defined in
+[schemaFileSchema.json](https://github.com/mozilla/telemetry-streaming/blob/master/src/main/resources/schemas/schemaFileSchema.json).
+We describe the meaning of each of the fields here.
+
+### source
+
+`source` is a string like `"source": "telemetry"` defining which raw ping dataset
+to use when the job runs in batch mode.
+Available datasets are listed in [sources.json](https://console.aws.amazon.com/s3/object/net-mozaws-prod-us-west-2-pipeline-metadata/sources.json?region=us-east-1&tab=overview) in S3.
+Note that each dataset contains a different set of fields it is partitioned by
+and performance for the job in batch mode can be greatly increased by configuring
+efficient filters on those partitioning fields in the `filters` section of the config.
+
+### filters
+
+`filters` is a map of fieldName to list of acceptable values. Consider:
+
+```json
+"docType": [
+  "main"
+]
+```
+
+This filter will ensure that we only consider pings where the `docType` is `main`.
+If we added additional entries to that list, they are treated as a logical OR, so we
+would consider events with any of the listed docTypes.
+In batch mode, filters will be applied not just based on the content of pings, but
+on partitioning fields. Configuring filters on partitioning fields of the source dataset
+means that we're able to greatly reduce the set of files that need to be read for the
+job to complete, significantly boosting performance.
+You can also configure filters for fields within the ping that aren't used for partitioning
+and in this case, we check the content of the ping after it has been read from disk;
+these filters apply to both batch and streaming mode.
+
+### eventGroups
+
+`eventGroups` is where the transformation logic is defined that determines what
+the events that make it to Amplitude look like. It has a more complex structure.
+
+Each group in this list has an `eventGroupName` and a list of events. The name of an
+event in Amplitude will be `eventGroupName - eventName`.
+
+The next section describes the structue of each `event` entry.
+
+### event
+
+`name` determines the name of the event as seen in Amplitude.
+
+`description` is a comment field. It doesn't get sent to Amplitude, but rather
+exists as an opportunity for you to document the purpose of the event or any
+additional caveats for future maintainters of the schema.
+
+`sessionIdOffset` is an optional field created for the `devTools` job that gives
+you a chance to identify a field to use to provide additional disambiguation
+between events.
```

Comment:
I don't really understand this field, so this description may be way off. cc @sunahsuh

Reply:
I like it. Will incorporate.